[RFC] vim-patch:8.0.0234,8.0.0236,8.0.0225

ckelsel opened the pull request:
#### vim-patch:8.0.0225: put in Visual block mode terminates early

Problem:    When a block is visually selected and put is used on the end of
the selection only one line is changed.
Solution:   Check for the end properly. (Christian Brabandt, neovim issue 5781)
                                                                                
https://github.com/vim/vim/commit/9957a10d0f0c34d8083af6ed66e198e4796038e0


#### vim-patch:8.0.0234: crash when using put in Visual mode

Problem:    When several lines are visually selected and one of them is short,
            using put may cause a crash. (Axel Bender)
Solution:   Check for a short line. (Christian Brabandt)

https://github.com/vim/vim/commit/941c12da3c087fd04aa6c120a76bf28f19349d96


#### vim-patch:8.0.0236: gcc complains about uninitialized variable

Problem:    Gcc complains that a variable may be used uninitialized. Confusion
            between variable and label name. (John Marriott)
Solution:   Initialize it.  Rename end to end_lnum.

https://github.com/vim/vim/commit/6a717f17ec6b09634be1c29e0ac4c35213f7b32d

ckelsel committed:
vim-patch:8.0.0234: crash when using put in Visual mode

Problem:    When several lines are visually selected and one of them is short,
            using put may cause a crash. (Axel Bender)
Solution:   Check for a short line. (Christian Brabandt)

https://github.com/vim/vim/commit/941c12da3c087fd04aa6c120a76bf28f19349d96

ckelsel committed:
vim-patch:8.0.0236: gcc complains about uninitialized variable

Problem:    Gcc complains that a variable may be used uninitialized. Confusion
            between variable and label name. (John Marriott)
Solution:   Initialize it.  Rename end to end_lnum.

https://github.com/vim/vim/commit/6a717f17ec6b09634be1c29e0ac4c35213f7b32d

ckelsel commented:
refer to #7999 vim-patch:8.0.0236: gcc complains about uninitialized variable

justinmk commented on src/nvim/ops.c:
```diff
@@ -3102,11 +3113,10 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
-               && (lnum <= curbuf->b_visual.vi_end.lnum
-                   || lnum <= curbuf->b_visual.vi_start.lnum));
+        }
+      } while (VIsual_active && lnum <= end_lnum);
```
Why not port 8.0.0225 first? It will be confusing to merge it later.

justinmk replied:
Sorry, I didn't realize we had already covered this in #5781. Please try to mention that if you can, it will save time.

ckelsel committed:
Merge remote-tracking branch 'upstream/master' into vim-8.0.0234

ckelsel committed:
vim-patch:8.0.0225: put in Visual block mode terminates early

Problem:    When a block is visually selected and put is used on the end of
            the selection only one line is changed.
Solution:   Check for the end properly. (Christian Brabandt, neovim issue
            5781)

https://github.com/vim/vim/commit/9957a10d0f0c34d8083af6ed66e198e4796038e0

ckelsel commented:
```
1690Error: Error fetching file: Failed downloading https://github.com/libmpack/libmpack-lua/releases/download/1.0.7/libmpack-lua-1.0.7.tar.gz - libmpack-lua-1.0.7.tar.gz
1691mingw32-make[2]: *** [CMakeFiles\mpack.dir\build.make:60: usr/lib/luarocks/rocks/mpack] Error 1
1692mingw32-make[2]: Leaving directory 'C:/projects/neovim/.deps'
1693mingw32-make[1]: *** [CMakeFiles\Makefile2:577: CMakeFiles/mpack.dir/all] Error 2
1694mingw32-make[1]: Leaving directory 'C:/projects/neovim/.deps'
1695mingw32-make: *** [Makefile:83: all] Error 2
1696
1697C:\projects\neovim\.deps>exit /b 2 
1698
```
